godot-core/src/builtin/variant/mod.rs

```diff
@@ -60,13 +60,29 @@ impl Variant {
 
     /// Checks whether the variant is empty (`null` value in GDScript).
     pub fn is_nil(&self) -> bool {
-        self.sys_type() == sys::GDEXTENSION_VARIANT_TYPE_NIL
+        // Use get_type rather than sys_type to include checking for the case where the variant is an object but with a null pointer.
+        self.get_type() == VariantType::Nil
     }
 
     // TODO test
+    /// Converts the sys_type to the VariantType.
+    /// Also handles the case where a variant's type is Object but it has a null pointer, instead returning Nil.
     pub fn get_type(&self) -> VariantType {
-        let ty_sys = unsafe { interface_fn!(variant_get_type)(self.var_sys()) };
-        VariantType::from_sys(ty_sys)
+        let sys_type = self.sys_type();
+        let is_a_null_object = sys_type == sys::GDEXTENSION_VARIANT_TYPE_OBJECT && {
+            unsafe {
+                let mut object_ptr: *const std::ffi::c_void = std::ptr::null();
+                let object_ptr_ptr: *mut crate::sys::TagType = std::mem::transmute(&mut object_ptr);
```

Comment:
I don't understand exactly what TagType is supposed to represent (and it didn't like what I think it's supposed to be, which is equivalent to `*mut GDExtensionObjectPtr`.
Someone else can probably point out what the right type coercion is here.

Reply:
`TypeTag` is simple used to make the `GDExtensionTypePtr` an opaque type, rather than `void*`.
This adds type safety: you cannot pass `GDExtensionTypePtr` where `GDExtensionVariantPtr` is expected.

Ideally Godot would implement that by themselves, but until then I use a patched header file.

Reply:
Regarding code, you should never mention `TagType` explicitly. Always use the typedefs like `sys::GDExtensionTypePtr`.

Reply:
Then what should I be passing to `converter` here?

Reply:
Also, I wouldn't use a huge `{}` block in a boolean expression, it costs nothing to store it in a variable 😉 

See comment in main thread; it's possible that this is no longer needed by reusing existing code.